src/state/slices/txHistorySlice/txHistorySlice.ts

```diff
@@ -5,13 +5,26 @@ export type Tx = chainAdapters.SubscribeTxsMessage<ChainTypes>
 
 const initialState = {} as { [key: string]: Tx[] }
 
+const updateSlice = (txs: Array<Tx>, tx: Tx): Array<Tx> => {
```

Comment:
Can you add a JSDoc block with a description of the purpose of this function?

This looks like an `upsert` action (update or insert), so I'd prefer not to use `update` as the function name and something more like `upsert` or `updateOrInsert`.

Is the "slice" intended to be sorted in any way? If not, `txs.push` is a much faster operation than building a new array each time.

Reply:
Will make a follow up PR for performance improvement